### DISCUSSION TITLE
Adds touch gestures to v6 (with custom build and event.js or by using hammer.js)

### DISCUSSION THREAD

**Discussion by HesamSoleymani** [fabricjs/fabric.js#9389] [Q&A]
Hi
In http://fabricjs.com/build/ we can include touch gestures for v5 currently. 
I just wonder if there is a similar thing for v6 ?

Also any help on where to start for adding multitouch support with hammer.js to v6 would be much appreciated 🙏

**Comment by ShaMan123** [comment]
You can look at the unmigarted file `src/canvas/canvas_gestures.mixin`...
Custom build is a dead concept in v6.
Subclassing will get you where you want.

@asturur we need to decide what to do about this...

**Reply by HesamSoleymani** [reply]
Thanks
You mean I should add some code (according to canvas_gestures.mixin.ts) to my project (rather than changing something in the fabric.js package source code itself) ?

import { scalingEqually } from '../controls/scale';
import { fireEvent } from '../controls/fireEvent';
import {
  degreesToRadians,
  radiansToDegrees,
} from '../util/misc/radiansDegreesConversion';
import { Canvas } from './Canvas';
import { CENTER } from '../constants';

canvas_gestures.mixin.ts has some imports from other files though...

**Reply by ShaMan123** [reply]
It has NOT been migrated
So indeed take parts of it as you need

**Reply by HesamSoleymani** [reply]
OK fine. 
But still... What about event.js? How should I add and use it in my project?

from canvas_gestures.mixin.ts:
   * @param {Event} e Event object by Event.js
   * @param {Event} self Event proxy object by Event.js
   
my question is about  "by event.js"  part

It seems event.js is not even a package on npm...

**Reply by asturur** [reply]
it mostly depends on what gestures you are interested in.
I find hammerJS better than eventjs, but the issue is that none of those are updated or maintained.

The gesture sitatuation is a bit in a stall for me, in my last touch project i did use hammer.js and i just reimplemented gesture through that ( i had only pinch to zoom and little else to take care of )

Defintely i don't want to marry to a touch library especially one that do not support pointer events natively.
